packages/comms/src/ecl/workunit.ts

```diff
@@ -1096,6 +1096,22 @@ export class Workunit extends StateObject<UWorkunitState, IWorkunitState> implem
         });
     }
 
+    publishEx(request: Partial<WsWorkunits.WsWorkunits.WUPublishWorkunit>) {
+        const service = new WsWorkunits.WorkunitsServiceEx({ baseUrl: "" });
+        const publishRequest = {
+            ...{
+                Wuid: this.Wuid,
+                Cluster: this.Cluster,
+                JobName: this.Jobname,
+                AllowForeignFiles: true,
+                Activate: 1,
+                Wait: 5000
+            },
+            ...request
+        };
```

Comment:
Could be shortened to:
```ts
        const publishRequest = {
            Wuid: this.Wuid,
            Cluster: this.Cluster,
            JobName: this.Jobname,
            AllowForeignFiles: true,
            Activate: 1,
            Wait: 5000,
            ...request
        } as WsWorkunits.WsWorkunits.WUPublishWorkunit;
        return service.WUPublishWorkunit(publishRequest);
```

Then you wouldn't need a `WUPublishWorkunitEx` overload?